File: app/models/case_history.py
from datetime import datetime
from sqlalchemy import Integer, String, ForeignKey, DateTime
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base

class CaseHistory(Base):
    __tablename__ = "case_history"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    case_id: Mapped[int] = mapped_column(Integer, ForeignKey("cases.id"), nullable=False, index=True)

    status: Mapped[str] = mapped_column(String(30), nullable=False)
    comment: Mapped[str] = mapped_column(String(255), nullable=False, default="")
    created_at: Mapped[datetime] = mapped_column(DateTime, nullable=False, default=datetime.utcnow)

File: veteran-aid-web/src/api/case-history.ts
import { http } from "@/api/http";

export type CaseHistoryItem = {
  id: number;
  case_id: number;
  status: string;
  comment: string;
  created_at: string; // ISO
};

export async function fetchCaseHistory(caseId: number) {
  const { data } = await http.get<CaseHistoryItem[]>(`/cases/${caseId}/history`);
  return data;
}
